shared/validate/validate.go

```diff
@@ -763,3 +763,46 @@ func IsAbsFilePath(value string) error {
 
 	return nil
 }
+
+func ParseNetworkVLANList(rawVLANValues []string) ([]string, error) {
+	var networkVLANList []string
+	for _, vlan := range rawVLANValues {
+		if err := IsNetworkVLAN(vlan); err != nil {
+			networkVLANList = append(networkVLANList, vlan)
+		} else if err := IsNetworkVLANRange(vlan); err != nil {
+			start, end, _ := ParseNetworkVLANRange(vlan)
+			for i := start; i <= end; i++ {
+				networkVLANList = append(networkVLANList, strconv.Itoa(i))
+			}
+		} else {
+			return nil, fmt.Errorf("Invalid VLAN/VLAN Range: %s", vlan)
+		}
+	}
+
+	return networkVLANList, nil
+}
+
+func ParseNetworkVLANRange(vlan string) (int, int, error) {
+	vlanRange := strings.Split(vlan, "-")
+	if len(vlanRange) != 2 {
+		return -1, -1, fmt.Errorf("Invalid VLAN range input: %s", vlan)
+	}
+
+	if IsNetworkVLAN(vlanRange[0]) != nil || IsNetworkVLAN(vlanRange[1]) != nil {
+		return -1, -1, fmt.Errorf("Invalid VLAN range boundary. start:%s, end:%s", vlanRange[0], vlanRange[1])
+	}
+
+	vlanRangeStart, _ := strconv.Atoi(vlanRange[0])
```

Comment:
@zakufish although we have validated it above, we should still handle the errors from these 2 lines in case anything changes in the future that means an invalid VLAN ID slips through. Thanks